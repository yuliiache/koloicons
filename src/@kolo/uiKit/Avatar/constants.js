export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
