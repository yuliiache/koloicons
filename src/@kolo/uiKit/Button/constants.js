export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
