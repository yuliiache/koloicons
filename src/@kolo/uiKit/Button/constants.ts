export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
