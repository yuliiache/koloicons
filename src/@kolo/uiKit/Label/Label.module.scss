@import "@kolo/design";

.label {
  display: inline-block;
  border-radius: 4px;
}

.default {
  color: $secFontColor;
  background-color: $DarkBG;
}

.success {
  color: $LabelSuccessColor;
  background-color: $LabelSuccessBGColor;
}

.primary {
  color: $LabelPrimaryColor;
  background-color: $LabelPrimaryBGColor;
}

.warning {
  color: $LabelWarningColor;
  background-color: $LabelWarningBGColor;
}

.small {
  padding: 4px 8px;
  @include main-text(0.6rem, 1.4, 400);
}

.large {
  padding: 8px;
  @include main-text(0.7rem, 1.2, 500);
}
