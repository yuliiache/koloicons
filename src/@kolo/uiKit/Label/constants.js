export const LABEL_TEST_ID = 'labelComponent';

export const LABEL_THEME = Object.freeze({
  DEFAULT: 'default',
  SUCCESS: 'success',
  PRIMARY: 'primary',
  WARNING: 'warning',
});

export const LABEL_SIZE = Object.freeze({
  SMALL: 'small',
  LARGE: 'large',
});
