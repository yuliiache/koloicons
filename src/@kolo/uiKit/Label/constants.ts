export const LABEL_TEST_ID = 'labelComponent';

export enum LabelTheme {
  DEFAULT = 'default',
  SUCCESS = 'success',
  PRIMARY = 'primary',
  WARNING = 'warning',
}

export enum LabelSize {
  SMALL = 'small',
  LARGE = 'large',
}
