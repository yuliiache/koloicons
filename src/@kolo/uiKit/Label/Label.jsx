import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import {LABEL_SIZE, LABEL_TEST_ID, LABEL_THEME} from './constants';

import styles from './Label.module.scss';

const Label = ({text, theme = LABEL_THEME.DEFAULT, size = LABEL_SIZE.SMALL}) => {
  const labelClassNames = classNames(styles.label, styles[`${theme}`], styles[`${size}`]);
  return (
    <span
      className={labelClassNames}
      data-testid={LABEL_TEST_ID}
    >
      {text}
    </span>
  );
};

Label.propTypes = {
  text: PropTypes.string.isRequired,
  theme: PropTypes.oneOf(Object.values(LABEL_THEME)),
  size: PropTypes.oneOf(Object.values(LABEL_SIZE)),
};

export default Label;
