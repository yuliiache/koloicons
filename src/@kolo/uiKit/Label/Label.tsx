import classNames from 'classnames';
import React, {FC} from 'react';

import {LABEL_TEST_ID, LabelSize, LabelTheme} from './constants';
import styles from './Label.module.scss';

interface LabelProps {
  text: string;
  theme?: LabelTheme;
  size?: LabelSize;
}

const Label: FC<LabelProps> = ({text, theme = LabelTheme.DEFAULT, size = LabelSize.SMALL}) => {
  const labelClassNames = classNames(styles.label, styles[`${theme}`], styles[`${size}`]);
  return (
    <span
      className={labelClassNames}
      data-testid={LABEL_TEST_ID}
    >
      {text}
    </span>
  );
};

export default Label;
