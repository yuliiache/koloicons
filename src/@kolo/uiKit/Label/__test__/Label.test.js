import React from 'react';
import {render} from '@testing-library/react';

import {LABEL_SIZE, LABEL_THEME} from '../constants';
import Label from '../Label';

describe('Label component', () => {
  Object.values(LABEL_THEME).forEach((theme) => {
    Object.values(LABEL_SIZE).forEach((size) => {
      it(`renders correctly for ${theme} style with ${size} size`, () => {
        const text = theme.charAt(0).toUpperCase() + theme.slice(1);
        const props = {
          text,
          theme,
          size,
        };
        const {asFragment} = render(<Label {...props} />);

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
