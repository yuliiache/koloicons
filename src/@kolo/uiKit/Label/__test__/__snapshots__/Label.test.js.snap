// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component renders correctly for default style with large size 1`] = `
<DocumentFragment>
  <span
    class="label default large"
    data-testid="labelComponent"
  >
    Default
  </span>
</DocumentFragment>
`;

exports[`Label component renders correctly for default style with small size 1`] = `
<DocumentFragment>
  <span
    class="label default small"
    data-testid="labelComponent"
  >
    Default
  </span>
</DocumentFragment>
`;

exports[`Label component renders correctly for primary style with large size 1`] = `
<DocumentFragment>
  <span
    class="label primary large"
    data-testid="labelComponent"
  >
    Primary
  </span>
</DocumentFragment>
`;

exports[`Label component renders correctly for primary style with small size 1`] = `
<DocumentFragment>
  <span
    class="label primary small"
    data-testid="labelComponent"
  >
    Primary
  </span>
</DocumentFragment>
`;

exports[`Label component renders correctly for success style with large size 1`] = `
<DocumentFragment>
  <span
    class="label success large"
    data-testid="labelComponent"
  >
    Success
  </span>
</DocumentFragment>
`;

exports[`Label component renders correctly for success style with small size 1`] = `
<DocumentFragment>
  <span
    class="label success small"
    data-testid="labelComponent"
  >
    Success
  </span>
</DocumentFragment>
`;

exports[`Label component renders correctly for warning style with large size 1`] = `
<DocumentFragment>
  <span
    class="label warning large"
    data-testid="labelComponent"
  >
    Warning
  </span>
</DocumentFragment>
`;

exports[`Label component renders correctly for warning style with small size 1`] = `
<DocumentFragment>
  <span
    class="label warning small"
    data-testid="labelComponent"
  >
    Warning
  </span>
</DocumentFragment>
`;
